src/main/webapp/script.js

```diff
@@ -0,0 +1,371 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+/**
+ * Function which is called when page loads.
+ * Add event listener to image upload button.
+ * Add canvases to the page, put sample image to them.
+ */
+document.addEventListener('DOMContentLoaded', () => {
+  const SAMPLE_IMAGE_URL = 'images/hadgehog.jpg';
+  const CANVAS_WIDTH = 300;
+
+  const uploadButton = document.getElementById('upload-image');
+  uploadButton.addEventListener('change', handleImageUpload);
+
+  const imagesContainer = document.getElementById('images-container');
+
+  // put canvases into image container on html page
+  const inputCanvas = createCanvas('input-canvas');
+  inputCanvas.setAttribute('width', CANVAS_WIDTH);
+  imagesContainer.append(inputCanvas);
+
+  const outputCanvas = createCanvas('output-canvas');
+  outputCanvas.setAttribute('width', CANVAS_WIDTH);
+  imagesContainer.append(outputCanvas);
+
+  // add sample image on page - original and blurred one.
+  blur(SAMPLE_IMAGE_URL);
+});
+
+/**
+ * Function which is called when user chooses file to upload.
+ * @param {Event} event Change event on image upload button.
+ */
+function handleImageUpload(event) {
+  if (!validateImageUpload()) {
+    return;
+  }
+
+  const imageUrl = URL.createObjectURL(event.target.files[0]);
+  blur(imageUrl);
+}
+
+/**
+ * Function to validate the uploaded image.
+ * @return {Boolean}
+ */
+function validateImageUpload() {
+  const fileName = document.getElementById('upload-image').value;
+
+  // if nothing was uploaded
+  if (fileName === '') {
+    alert('We can blur anything only if you upload an image');
+    return false;
+  }
+
+  // validate file extension
+  const SUPPORTED_EXTENSIONS = ['png', 'jpeg', 'jpg'];
+  const fileExtension = fileName.substring(
+      fileName.lastIndexOf('.') + 1).toLowerCase();
+
+  if (!SUPPORTED_EXTENSIONS.includes(fileExtension)) {
+    alert('We can support only jpg, jpeg and png images, sorry :(');
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Creates canvas with specified id.
+ * @param {string} id
+ * @return {HTMLCanvasElement} canvas DOM element
+ */
+function createCanvas(id) {
```

Comment:
Would be better to call it createCanvasForId, for example - to make it symmetric to createCanvasForImage.